### DISCUSSION TITLE
Support for commonjs in @auth/express

### DISCUSSION THREAD

**Discussion by noam-honig** [nextauthjs/next-auth#10507] [Ideas]
### Goals

I have many applications that use nodejs common js, and converting them to esm will require massive work - would be great if commonjs was supported too

### Non-Goals

I have many applications that use nodejs common js, and converting them to esm will require massive work - would be great if commonjs was supported too

### Background

I have many applications that use nodejs common js, and converting them to esm will require massive work - would be great if commonjs was supported too

### Proposal

I have many applications that use nodejs common js, and converting them to esm will require massive work - would be great if commonjs was supported too

**Comment by hillac** [comment]
The underlying @auth/core is ESM only and it's not worth changing that. You can always try use [tsx](https://tsx.is) with [pkgroll](https://tsx.is/compilation) to compile, that can apparently handle CJS and ESM.
